### PR TITLE
NMS-13584: Notification Commands Documentation

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13584

### CODE REVIEW

**Comment by Bonrob2** [comment]
@mmahacek I am unable to assign you to review this, as you were the original creator of the PR. Note that I was unable to view the Wiki content for the How To Create Notifications, as it seems to be down. However, we already had a procedure that covered that information through the UI. Let me know if we are missing anything.

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 1]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
 . Click *Finish*.
 
 NOTES: The event notification configuration is stored in the `notifications.xml` file.
 Changes to this file are picked up on the fly with no need to restart or send an event.
 
-The filter rule configured in notifications.xml, e.g., `<rule>IPADDR != '0.0.0.0'</rule>` is not strict by default.
-That means if there is any event that is not associated with any node/interface, it would not validate rule and by default notification would be saved.
-A strict rule e.g.,
-`<rule strict="true">IPADDR != '0.0.0.0'</rule>` will always be evaluated and if there is no node/interface associated with event, the notification would not be saved.
+The filter rule configured in notifications.xml (for example, `<rule>IPADDR != '0.0.0.0'</rule>`) is not strict by default.
```
Comment (mmahacek) [inline docs/modules/operation/pages/notifications/getting-started.adoc:57]:
```suggestion
The filter rule configured in `notifications.xml` (for example, `<rule>IPADDR != '0.0.0.0'</rule>`) is not strict by default.
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 2]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
-You can configure OpenNMS to execute only the destination path of the first matching notification by editing the `notifd-configuration.xml` file and setting `match-all="false"` in the top-level `notifd-configuration` element.
+By default, {page-component-title} runss the destination path of all notifications matching the event's UEI.
+You can configure {page-component-title} to run only the destination path of the first matching notification by editing the `notifd-configuration.xml` file and setting `match-all="false"` in the top-level `notifd-configuration` element.
 This configuration file change happens immediately with no need to restart or send an event.
 
-== Configuring Path Outages
+== Configure path outages
 
-Configuring a path outage lets you suppress notifications for nodes that appear to be down due to a failure in the network path between the nodes and OpenNMS.
+Configuring a path outage lets you suppress notifications for nodes that appear to be down due to a failure in the network path between the nodes and  {page-component-title}.
```
Comment (mmahacek) [inline docs/modules/operation/pages/notifications/getting-started.adoc:67]:
```suggestion
Configuring a path outage lets you suppress notifications for nodes that appear to be down due to a failure in the network path between the nodes and {page-component-title}.
```

**Comment by Bonrob2** [comment]
@mmahacek updated with your feedback. Requested review from Emily.

File: docs/modules/operation/pages/notifications/commands.adoc [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                <switch>trapSpecific</switch>
+        </argument>
+        <argument streamed="false">
+                <switch>trapVarbind</switch>
+        </argument>
+    </command>
+----
+
+You can define any or none of these parameters and switches.
+However, if you define it as a parameter in the notification, you must define it as a switch in the notification command.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/commands.adoc:239]:
```suggestion
However, if you define a parameter in the notification, you must also define a corresponding switch in the notification command.
```

File: docs/modules/operation/pages/notifications/concepts.adoc [hunk 4]
```diff
@@ -111,39 +111,3 @@ If multiple commands are selected, they will execute simultaneously.
 Each step also has an auto-notify switch, which may be set to `off`, `on`, or `auto`.
 This switch specifies the logic used when deciding whether or not to send a notice for an auto-acknowledged notification to a target that was not on duty at the time the notification was first created.
 If `off`, notices will never be sent to such a target; if `on`, they will always be sent; if `auto`, the system employs heuristics aimed at "doing the right thing".
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/concepts.adoc:113]:
```suggestion
If `off`, notices will never be sent to such a target; if `on`, they will always be sent; if `auto`, the system employs heuristics aimed at "doing the right thing."
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 5]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
 NOTE: The web workflow above is functionally equivalent to editing the `notifd-configuration.xml` file and setting `status="on"` in the top-level `notifd-configuration` element.
 This configuration file change happens immediately with no need to restart or send an event.
 
-== Configuring Destination Paths
+== Configure destination paths
 
 A destination path specifies the "who" (recipient), "when" (interval), and "how" (method) of the notification.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:25]:
```suggestion
A destination path specifies the "who" (recipient), "when" (interval), and "how" (method) of a notification.
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 6]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
 It also specifies targets for escalation.
 
 Configuring destination paths separately from individual events lets you use the same information for multiple notifications, minimizing duplication and encouraging reuse.
 
 . Log in to the web UI as a user with administrator privileges.
-. Click the gear icon in the top right, and under *Event Management* choose *Configure Notifications>Configure Destination Paths*.
+. Click the gear icon in the top right, and choose menu:Event Management[Configure Notifications > Configure Destination Paths].
 . Click *New Path*, fill in the desired information and click *Finish*.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:32]:
```suggestion
. Click *New Path*, fill in the desired information, and click *Finish*.
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 7]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
 
-An event notification ties any OpenNMS event with a destination path.
+An event notification ties any  {page-component-title} event with a destination path.
 
 . Log in to the web UI as a user with administrator privileges.
-. Click the gear icon in the top right, and under *Event Management* choose *Configure Notifications>Configure Event Notifications*.
+. Click the gears icon in the top right, and choose menu:Event Management[Configure Notifications>Configure Event Notifications].
 
 +
 The screen displays a list of events configured for notifications.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:46]:
```suggestion
+
The screen displays a list of events configured for notifications.
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 8]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
 
 . Click *Add New Event Notification*.
 . Choose an event on which you want to receive a notification and click *Next*.
 . Optionally build a rule to match a subset of IP addresses or services and click *Validate rule results*, then *Next* (or *Skip results validation*).
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:50]:
```suggestion
. (Optional) Build a rule to match a subset of IP addresses or services and click *Validate rule results*, then *Next* (or *Skip results validation*).
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 9]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
-. In the *Choose Path* screen, specify information to include in the notification, including destination path, subject (for an email message), text, etc. 
+. In the *Choose Path* screen, specify information to include in the notification, including destination path, subject (for an email message), text, and so on.
 . Click *Finish*.
 
 NOTES: The event notification configuration is stored in the `notifications.xml` file.
 Changes to this file are picked up on the fly with no need to restart or send an event.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:55]:
```suggestion
NOTE: The event notification configuration is stored in the `notifications.xml` file.
Changes to this configuration file happen immediately with no need to restart or send an event.
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 10]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
 
-The filter rule configured in notifications.xml, e.g., `<rule>IPADDR != '0.0.0.0'</rule>` is not strict by default.
-That means if there is any event that is not associated with any node/interface, it would not validate rule and by default notification would be saved.
-A strict rule e.g.,
-`<rule strict="true">IPADDR != '0.0.0.0'</rule>` will always be evaluated and if there is no node/interface associated with event, the notification would not be saved.
+The filter rule configured in `notifications.xml` (for example, `<rule>IPADDR != '0.0.0.0'</rule>`) is not strict by default.
+That means if there is any event that is not associated with any node or interface, it would not validate the rule and by default notification would be saved.
+A strict rule (for example, `<rule strict="true">IPADDR != '0.0.0.0'</rule>`) will always be evaluated and if there is no node or interface associated with the event, the notification would not be saved.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:59]:
```suggestion
A strict rule (for example, `<rule strict="true">IPADDR != '0.0.0.0'</rule>`) will always be evaluated, and if there is no node or interface associated with the event, the notification would not be saved.
```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 11]
```diff
@@ -20,57 +20,56 @@ The green bell icon indicates that notifications are globally enabled.
...
 
-By default, OpenNMS executes the destination path of all notifications matching the event's UEI.
-You can configure OpenNMS to execute only the destination path of the first matching notification by editing the `notifd-configuration.xml` file and setting `match-all="false"` in the top-level `notifd-configuration` element.
+By default, {page-component-title} runss the destination path of all notifications matching the event's UEI.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:61]:
```suggestion
By default, {page-component-title} runs the destination path of all notifications matching the event's UEI.
```

File: docs/modules/operation/pages/notifications/commands.adoc [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+
+=== Extended dynamic values
+
+You can pass arbitrary parameters from the notification to the notificationCommand via the <parameter> tag, which can then be used as required.
+
+== Examples
+
+=== SnmpTrapNotificationStrategy
+
+This configuration sends an OpenNMS trap to an external system with the `nodelabel` in the varbind.
```
[the remark this answers is not in the document]
Comment (Bonrob2) [reply docs/modules/operation/pages/notifications/commands.adoc:174]: Let's use OpenNMS.

**Review by mmahacek — COMMENTED** [review]
I can't approve since I created the PR, but LGTM.

File: docs/modules/operation/pages/notifications/commands.adoc [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+trapHost="127.0.0.1" (any valid hostname or ip address)
+trapPort="162" (any valid IP port)
+trapCommunity="public" (any valid community string)
+trapEnterprise=".1.3.6.1.4.1.5813" (any valid OID)
+trapGeneric="6" (any valid generic ID)
+trapSpecific="1" (any valid trapSpecific ID)
+trapVarbind="OpenNMS Trap Notification" (any string or one or more valid %<event field>% such as "%eventUEI%" in the string)
+----
+
+The one allowed trap varbind will be sent with the object ID `.1.3.6.1.4.1.5813.20.1` and the object type `DisplayString`.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/commands.adoc:255]:
```suggestion
The one allowed trap varbind will be sent with the object ID `.1.3.6.1.4.1.5813.20.1` and the object type `DisplayString`.

```

File: docs/modules/operation/pages/notifications/concepts.adoc [hunk 14]
```diff
@@ -110,40 +110,4 @@ If multiple commands are selected, they will execute simultaneously.
...
-`xmppGroupMessage` and `xmppMessage`::
-    Sends a message to an XMPP group or user.
-    Implemented with the in-process XMPP notification strategy.
-
-Notification commands are customizable and extensible by editing the `notificationCommands.xml` file.
-
-NOTE: Use external binary notification commands sparingly to avoid fork-bombing your {page-component-title} system.
-Originally, all notification commands were external.
-Today only the `numericPage` and `textPage` commands use external programs to do their work.
+If `off`, notices will never be sent to such a target; if `on`, they will always be sent; if `auto`, the system employs heuristics aimed at "doing the right thing."
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/concepts.adoc:113]:
```suggestion
If `off`, notices will never be sent to such a target; if `on`, they will always be sent; if `auto`, the system employs heuristics aimed at "doing the right thing."

```

File: docs/modules/operation/pages/notifications/getting-started.adoc [hunk 15]
```diff
@@ -20,57 +20,55 @@ The green bell icon indicates that notifications are globally enabled.
...
 For example, if a WAN link fails, all nodes at the remote site served by the WAN link will appear to be down.
 Since we will get a notification that the router on the far end of the WAN link is not responding, we don't need notifications for all the devices that sit behind that router.
 
 You can configure a path outage for an individual node or create a rule-based configuration for a group of nodes:
 
 . Log in to the web UI as a user with administrator privileges.
-. Click the gear icon in the top right, and under *Event Management* choose *Configure Notifications>Configure Path Outages*.
-. Type the critical path IP address, select a critical path service, and click *Next*.
+. Click the gear icon in the top right, and choose menu:Event Management[Configure Notifications>Configure Path Outages].
+. Type the critical path IP address, select a critical path service, and click *Next*.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/notifications/getting-started.adoc:74]:
```suggestion
. Type the critical path IP address, select a critical path service, and click *Next*.

```

**Review by emarsh1995 — APPROVED** [review]
Just adding blank lines to the ends of the files. Otherwise, LGTM. Thanks!

**Review by Bonrob2 — DISMISSED** [review]
LGTM!

**Review by emarsh1995 — APPROVED** [review]
LGTM!

**Pull request merged**